libmamba/src/core/solver.cpp

```diff
@@ -241,29 +241,59 @@ namespace mamba
         }
     }
 
-    void MSolver::set_postsolve_flags(const std::vector<std::pair<int, int>>& flags)
+    void MSolver::py_set_postsolve_flags(const std::vector<std::pair<int, int>>& flags)
     {
         for (const auto& option : flags)
         {
             switch (option.first)
             {
-                case MAMBA_NO_DEPS:
-                    no_deps = option.second;
+                case PY_MAMBA_NO_DEPS:
+                    m_flags.keep_dependencies = !option.second;
                     break;
-                case MAMBA_ONLY_DEPS:
-                    only_deps = option.second;
+                case PY_MAMBA_ONLY_DEPS:
+                    m_flags.keep_specs = !option.second;
                     break;
-                case MAMBA_FORCE_REINSTALL:
-                    force_reinstall = option.second;
+                case PY_MAMBA_FORCE_REINSTALL:
+                    m_flags.force_reinstall = option.second;
                     break;
             }
         }
     }
 
-    void MSolver::set_flags(const std::vector<std::pair<int, int>>& flags)
+    void MSolver::set_flags(const Flags& flags)
+    {
+        m_flags = flags;
+    }
+
+    auto MSolver::flags() const -> const Flags&
+    {
+        return m_flags;
+    }
+
+    auto MSolver::flags() -> Flags&
+    {
+        return m_flags;
+    }
+
+    void MSolver::set_libsolv_flags(const std::vector<std::pair<int, int>>& flags)
+    {
+        m_libsolv_flags = flags;
+    }
+
+    auto MSolver::libsolv_flags() const -> const std::vector<std::pair<int, int>>&
+    {
+        return m_libsolv_flags;
+    }
+
+    auto MSolver::libsolv_flags() -> std::vector<std::pair<int, int>>&
+    {
+        return m_libsolv_flags;
+    }
+
+    void MSolver::apply_libsolv_flags()
     {
         // TODO use new API
```

Comment:
Is this comment still relevant?

Reply:
Probably. The solver flags API will be reworked with the new Solver API.